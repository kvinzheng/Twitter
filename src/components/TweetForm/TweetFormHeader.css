.TweetFormHeader{
  display: flex;
  width: 100%;
  min-height: 45px;
  max-height: 125px;
  overflow-y: auto;
  border: solid #dee1e1;
  border-width: 0 0 1px;
  cursor: pointer;
}
.TweetFormHeader-detail {
  display: flex;
  align-items: center;
  justify-content: center;
}
.TweetFormHeader-name {
  margin: 0px 4px 0px 0px;
  color: rgb(54, 65, 65);
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
}
.TweetFormHeader-account {
  margin: 0px 2px;
  color: rgb(110, 121, 122);
  font-size: 13px;
  line-height: 21.3333px;
}
