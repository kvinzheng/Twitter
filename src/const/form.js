export const TEXT_AREA_COUNT = 280;
export const FETCH_DELAY_TIME = 1000;