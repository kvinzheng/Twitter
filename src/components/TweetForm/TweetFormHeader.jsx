import React from "react";
import { Twitter } from "react-feather";

import "./TweetFormHeader.css";

const TweetFormHeader = () => (
  <div className="TweetFormHeader">
    <img src="https://pbs.twimg.com/profile_images/1186398821560897536/1Vp8_at4_normal.jpg" />
    <div className="TweetFormHeader-detail">
      <Twitter
        size="20"
        fill="rgba(29,161,242,1.00)"
        className="TweetFormHeader-twitter-logo"
      />
      <span className="TweetFormHeader-name">Your Name</span>
      <span className="TweetFormHeader-account">@yourname</span>
    </div>
  </div>
);

export default TweetFormHeader;
