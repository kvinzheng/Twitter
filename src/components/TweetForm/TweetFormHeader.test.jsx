import React from "react";
import { shallow, configure, mount } from "enzyme";
import { TEXT_AREA_COUNT } from "../../const/form";
import TweetFormHeader from "./TweetFormHeader.jsx";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("TweetFormHeader Component", () => {
  it("should render without any data defined", () => {
    const component = shallow(<TweetFormHeader />);
    expect(component).toMatchSnapshot();
  });

  it("Renders with correct classes", () => {
    const component = shallow(
      <TweetFormHeader
        error={
          "please editing an @ followed by 2 alphanumeric characters (a-z, 0-9)"
        }
        status={TEXT_AREA_COUNT}
      />
    );
    expect(component.find(".TweetFormHeader").length).toBe(1);
    expect(component.find(".TweetFormHeader-detail").length).toBe(1);
    expect(component.find(".TweetFormHeader-twitter-logo").length).toBe(1);
    expect(component.find(".TweetFormHeader-name").length).toBe(1);
    expect(component.find(".TweetFormHeader-account").length).toBe(1);
  });
});
