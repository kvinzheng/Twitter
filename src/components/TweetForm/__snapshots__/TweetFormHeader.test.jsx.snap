// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetFormHeader Component should render without any data defined 1`] = `ShallowWrapper {}`;
